fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneScanAllEntriesTest.java

```diff
@@ -0,0 +1,220 @@
+                Arguments.of(true, true, true, false, Set.of(syntheticKeyFor(2, 1001L, 1000L))), // synthetic, ALL_ENTRIES, grouped, no EmptyDoc
+                Arguments.of(true, true, false, true, Set.of(syntheticKeyFor(1, 1001L, 1000L), syntheticKeyFor(2, 1002L, 999L), syntheticKeyFor(2, 1003L, 998L))), // synthetic, ALL_ENTRIES, non-grouped, with EmptyDoc
+                Arguments.of(true, true, false, false, Set.of(syntheticKeyFor(1, 1001L, 1000L), syntheticKeyFor(2, 1002L, 999L))) // synthetic, ALL_ENTRIES, non-grouped, no EmptyDoc
+        );
+    }
+
+    @ParameterizedTest(name = "indexScanTest({argumentsWithNames})")
+    @MethodSource("scanArguments")
+    public void indexScanTest(boolean isSynthetic, boolean matchAllDocs, boolean isGrouped, boolean includeEmptyDoc, Set<Tuple> expectedResult) throws Exception {
+        final long seed = 5363275763521L;
+        final LuceneIndexTestDataModel dataModel = new LuceneIndexTestDataModel.Builder(seed, this::getStoreBuilder, pathManager)
+                .setIsGrouped(isGrouped)
+                .setIsSynthetic(isSynthetic)
+                .setPrimaryKeySegmentIndexEnabled(true)
+                .setPartitionHighWatermark(10)
+                .build();
+
+        final long start = Instant.now().toEpochMilli();
+        // Populate data: 1 doc for groups 1 and 2 and one empty doc (if needed)
+        try (FDBRecordContext context = openContext()) {
+            dataModel.saveRecords(1, start, context, 1);
+            dataModel.saveRecords(1, start, context, 2);
+            if (includeEmptyDoc) {
+                LuceneIndexTestDataModel.saveEmptyRecord(isGrouped, isSynthetic, dataModel.random, dataModel.groupingKeyToPrimaryKeyToPartitionKey, start, dataModel.createOrOpenRecordStore(context), 2);
+            }
+            commit(context);
+        }
+
+        LuceneQueryClause search = matchAllDocs
+                                   ? LuceneQuerySearchClause.MATCH_ALL_DOCS_QUERY
+                                   : isSynthetic
+                                     ? new LuceneQuerySearchClause(LuceneQueryType.QUERY, CHILD_SEARCH_TERM, false)
+                                     : new LuceneQuerySearchClause(LuceneQueryType.QUERY, PARENT_SEARCH_TERM, false);
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = dataModel.createOrOpenRecordStore(context);
+            LuceneScanBounds scanBounds = isGrouped
+                                          ? LuceneIndexTestValidator.groupedSortedTextSearch(store, dataModel.index, search, null, 2)
+                                          : LuceneIndexTestUtils.fullTextSearch(store, dataModel.index, search, false);
+
+            // Run the scan with the given query and assert the results
+            assertIndexEntryPrimaryKeyTuples(expectedResult,
+                    store.scanIndex(dataModel.index, scanBounds, null, ScanProperties.FORWARD_SCAN));
+        }
```

Comment:
Other than the `expectedResult` these look to be the same in all tests, so I think a helper method would clean that up.

Reply:
All test methods now use the same structure of matching to a Set<Tuple>